pkg/operator/controller/garden/garden/reconciler_delete.go

```diff
@@ -445,3 +453,34 @@ func (r *Reconciler) destroyGardenPrometheus(ctx context.Context, prometheus pro
 
 	return r.RuntimeClientSet.Client().DeleteAllOf(ctx, &corev1.Secret{}, client.InNamespace(r.GardenNamespace), client.MatchingLabels{v1beta1constants.GardenerPurpose: gardenerutils.LabelPurposeGlobalMonitoringSecret})
 }
+
+func (r *Reconciler) destroyDNSRecords(ctx context.Context, log logr.Logger, garden *operatorv1alpha1.Garden) error {
+	if garden.Spec.VirtualCluster.DNS.Provider == nil || garden.Spec.VirtualCluster.DNS.SecretRef == nil {
+		return fmt.Errorf("no DNS provider or DNS secret configuration found in Garden resource")
+	}
+
+	dnsRecordList := &extensionsv1alpha1.DNSRecordList{}
+	if err := r.RuntimeClientSet.Client().List(ctx, dnsRecordList, client.InNamespace(r.GardenNamespace)); err != nil {
+		return fmt.Errorf("failed listing DNS records: %w", err)
```

Comment:
I think we should only consider deleting the records that we once created or managed. What do you think about working with a label to identify those?

Reply:
introduced label `origin=gardener-operator`